docs/dev-guide/contributing/bugs.rst

```diff
@@ -80,6 +80,15 @@ Fixing
 
 When fixing a bug, all bugs will follow this process, regardless of how trivial.
 
+First of all you need to reporoduce this bug. If this error is connected with client try to use
+``-vv`` flag to obtain complete information. Also it can be useful to look into server log.
+If there is traceback try to look first there and individualy look for bug. In some cases you can
+try to reproduce the error in python console. Be aware of fact that bug can not only be caused by
+just plugin or platform. Once you know where the problem is try to fix it. Always write unittests.
+Before you commit manualy test if bug is fixed after that you should run all unittests via
+``run-tests.py`` to check whether something got broken by your fix. If tests pass
+you are good to push and submit PR by rules mentioned below.
+
```

Comment:
I think you can take out the sentence "once you know where the problem is, try to fix it", and expand the "always write unittests" sentence to something like "always fix any unittests broken by a change, and add new test code to cover code additions."